src/list/item/ListItem.tsx

```diff
@@ -0,0 +1,73 @@
+import "./_list-item.scss";
+
+import React from "react";
+import classNames from "classnames";
+
+import {KEYBOARD_EVENT_KEY} from "../../core/utils/keyboardEventConstants";
+
+interface ListItemProps {
+  children?: React.ReactNode;
+  testid?: string;
+  customClassName?: string;
+  clickableListItemProps?: {
+    onClick: React.ReactEventHandler;
+    tabIndex?: number;
+  };
+  id?: string;
+  role?: string;
+  ariaSelected?: boolean;
+  listItemRef?: React.RefObject<HTMLLIElement>;
+}
+
+function ListItem({
+  id,
+  testid,
+  children,
+  customClassName,
+  clickableListItemProps,
+  role,
+  ariaSelected,
+  listItemRef
+}: ListItemProps) {
+  const containerClassName = classNames("list-item", customClassName);
+  let listItem = (
+    <li
+      ref={listItemRef}
+      id={id}
+      data-testid={testid}
+      className={containerClassName}
+      role={role}
+      aria-selected={ariaSelected}>
+      {children}
+    </li>
+  );
+
+  if (clickableListItemProps) {
+    listItem = (
+      <li ref={listItemRef} id={id} data-testid={testid} className={containerClassName}>
+        <div
+          role={"button"}
+          tabIndex={clickableListItemProps.tabIndex || 0}
+          className={"list-item__click-wrapper"}
+          onClick={handleClick}
+          onKeyPress={handleKeyPress}>
+          {children}
+        </div>
+      </li>
+    );
+  }
+
+  return listItem;
```

Comment:
```suggestion
  return clickableListItemProps ? (
        <li ref={listItemRef} id={id} data-testid={testid} className={containerClassName}>
        <div
          role={"button"}
          tabIndex={clickableListItemProps.tabIndex || 0}
          className={"list-item__click-wrapper"}
          onClick={handleClick}
          onKeyPress={handleKeyPress}>
          {children}
        </div>
      </li>
    
  ) : (
    <li
      ref={listItemRef}
      id={id}
      data-testid={testid}
      className={containerClassName}
      role={role}
      aria-selected={ariaSelected}>
      {children}
    </li>
    );
  }
```
I find this version more readable and meaningful, because we have only two variant and a single condition for choosing

Reply:
I did use ternary expression in the first place but then I decided to go with the other method. But I can change it back to the ternary, of course. 😄